beets/util/artresizer.py

```diff
@@ -92,11 +93,13 @@ def im_resize(maxwidth, path_in, path_out=None):
     # than the given width while maintaining the aspect ratio
     # with regards to the height.
     try:
-        util.command_output([
-            'convert', util.syspath(path_in, prefix=False),
-            '-resize', '{0}x>'.format(maxwidth),
-            util.syspath(path_out, prefix=False),
-        ])
+        cmds = (['magick'], ['convert'])
+        cmd = cmds[0] if not ArtResizer.shared.im_legacy else cmds[1]
```

Comment:
Or, even better than the above: consider making the `ArtResizer` set a `convert_cmd` and `identify_cmd` field on itself, which chooses between these two options. Then the code that uses the command doesn't need to duplicate the logic to check `ArtResizer.shared.im_legacy` and switch between the two options.

Reply:
Added in 48be3a7